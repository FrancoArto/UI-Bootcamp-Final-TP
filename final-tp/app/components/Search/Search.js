'use strict';

import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { Content, Item, Button, Input, Icon } from 'native-base';
import styles from './Search.style';


class Search extends Component {
    constructor(props) {
        super(props);

        this.onChangeText = this.onChangeText.bind(this);
        this.handleOnPress = this.handleOnPress.bind(this);
        this.handleClearPress = this.handleClearPress.bind(this);
    }
    
    handleOnPress() {
      let search = this.props.searchText;
      search = search.replace(/#/, '%23');
      this.props.onSearch(search);
    }

    onChangeText(value) {
      this.props.onChangeText(value);
    }
    
    handleClearPress() {
      this.props.onClearPress();
    }

    render() {
      let icon
      if (this.props.searchText !== '') {
        icon = <Icon active name='close' style={{color: 'black'}} onPress={this.handleClearPress} />              
      } 
        return(
          <View style={styles.container}>
            <Content>
              <Item style={styles.hideBottomBorder}>
                <Item style={styles.input}> 
                  <Icon active name='search' style={{color: 'lightskyblue'}} />
                  <Input value={this.props.searchText} onChangeText={this.onChangeText} />
                  {icon}
                </Item>
                <Item style={[styles.searchButton, styles.hideBottomBorder]}>
                  <Button info onPress={this.handleOnPress}>
                    <Text style={styles.searchButtonText}>Search</Text>
                  </Button>                
                </Item>
              </Item>             
            </Content>
          </View>
        );
    }
};

export default Search;