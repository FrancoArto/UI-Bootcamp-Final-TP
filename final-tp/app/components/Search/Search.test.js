import React from 'react';
import {mount, shallow} from 'enzyme'
import Search from './Search';


it('renders without crashing', () => {
  const rendered = shallow(<Search />)
  expect(rendered).toMatchSnapshot();
})

it('updates search text', () => {
  var searchText = ''
  function onChangeText (value) {
    searchText = value
    return value
  } 
  const wrapper = mount(<Search 
    onChangeText={onChangeText}
    searchText={searchText}/>)

  searchText = 'Hola'
  const textInput = wrapper.find('Input').first();
  textInput.props().onChangeText(searchText);
  wrapper.setProps({
    searchText: searchText
  })
  wrapper.update()
  expect(wrapper.props().searchText).toEqual(searchText)
});

it('clears search text', () => {
  let searchText = 'Hola'
  function onClearPress () {
    searchText = ''
  } 
  const wrapper = mount(<Search 
    searchText={searchText}
    onClearPress={onClearPress}/>)
 
  const clearIcon = wrapper.find('Icon').last();
  clearIcon.props().onPress()
  wrapper.setProps({
    searchText: searchText
  })
  wrapper.update()
  expect(wrapper.props().searchText).toEqual(searchText)
});

it('dispatch search', () => {
  let searchText = 'Hola'
  const onSearch = jest.fn()
  
  const wrapper = mount(<Search 
    searchText={searchText}
    onSearch={onSearch}/>)
 
  const searchButton = wrapper.find('Button').first();
  searchButton.props().onPress()
  expect(onSearch).toHaveBeenCalled()
});