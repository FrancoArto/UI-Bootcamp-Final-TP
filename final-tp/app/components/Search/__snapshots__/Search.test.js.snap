// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Component
  style={
    Object {
      "flex": 0.15,
      "flexDirection": "row",
    }
  }
>
  <Styled(Content)>
    <Styled(Item)
      style={
        Object {
          "borderBottomWidth": 0,
        }
      }
    >
      <Styled(Item)
        style={
          Object {
            "borderBottomWidth": 0.5,
            "borderColor": "black",
            "borderLeftWidth": 0.5,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "height": "80%",
            "marginLeft": 10,
            "marginTop": 5,
            "paddingLeft": 5,
            "width": "70%",
          }
        }
      >
        <Styled(Icon)
          active={true}
          name="search"
          style={
            Object {
              "color": "lightskyblue",
            }
          }
        />
        <Styled(Input)
          onChangeText={[Function]}
        />
        <Styled(Icon)
          active={true}
          name="close"
          onPress={[Function]}
          style={
            Object {
              "color": "black",
            }
          }
        />
      </Styled(Item)>
      <Styled(Item)
        style={
          Array [
            Object {
              "height": "80%",
              "justifyContent": "center",
              "marginTop": 5,
              "paddingRight": 10,
              "width": "30%",
            },
            Object {
              "borderBottomWidth": 0,
            },
          ]
        }
      >
        <Styled(Button)
          info={true}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "marginHorizontal": 20,
                "textAlign": "center",
              }
            }
          >
            Search
          </Text>
        </Styled(Button)>
      </Styled(Item)>
    </Styled(Item)>
  </Styled(Content)>
</Component>
`;
